### ISSUE TITLE
infinite loop

### ISSUE THREAD

**Issue by keeganstothert** [ghengeveld/react-async#80]
```js
function DeleteScheduleForm() {
  const { someCondition, setSomeCondition} = useState();
  const { isLoading, run } = useAsync({
    deferFn: deleteSchedule,
  });

  useEffect(() => {
    if(!someCondition) return 

    run();
  }, [someCondition, run]);

  // ...
}

```

the useEffect requires run in the dependencies but this seems to be the cause of an infinite loop. the issue goes away if i ignore eslint and remove run from the list of dependencies

**Comment by ghengeveld** [comment]
Thanks for reporting. I was able to create [a sandbox](https://codesandbox.io/s/great-gould-vb28r) and unit test to reproduce the behavior. I've also created a fix by wrapping `run` with `useCallback`. This seems to be sufficient.

**Comment by ghengeveld** [comment]
Released in [v8.0.0](https://github.com/ghengeveld/react-async/releases/tag/v8.0.0)
